src/internet/radiobrowser/radiobrowserservice.cpp

```diff
@@ -0,0 +1,390 @@
+
+void RadioBrowserService::Homepage() {
+  QDesktopServices::openUrl(homepage_url_);
+}
+
+PlaylistItem::Options RadioBrowserService::playlistitem_options() const {
+  return PlaylistItem::PauseDisabled | PlaylistItem::SeekDisabled;
+}
+
+void RadioBrowserService::PopulateCategory(QStandardItem* parentItem,
+                                           QStringList& elements) {
+  if (parentItem->hasChildren())
+    parentItem->removeRows(0, parentItem->rowCount());
+
+  for (const QString& element : elements) {
+    QStandardItem* item = new QStandardItem(
+        IconLoader::Load("icon_radio", IconLoader::Lastfm), QString());
+    item->setText(element);
+    item->setData(RadioBrowserService::Type_CategoryItem,
+                  InternetModel::Role_Type);
+    item->setData(true, InternetModel::Role_CanLazyLoad);
+    parentItem->appendRow(item);
+  }
+}
+
+void RadioBrowserService::PopulateStreams(QStandardItem* parentItem,
+                                          StreamList& streams) {
+  if (parentItem->hasChildren())
+    parentItem->removeRows(0, parentItem->rowCount());
+
+  for (const Stream& stream : streams) {
+    QStandardItem* item = new QStandardItem(
+        IconLoader::Load("icon_radio", IconLoader::Lastfm), QString());
+    item->setText(stream.name_);
+    item->setData(QVariant::fromValue(stream.ToSong(name_)),
+                  InternetModel::Role_SongMetadata);
+    item->setData(InternetModel::PlayBehaviour_SingleItem,
+                  InternetModel::Role_PlayBehaviour);
+
+    parentItem->appendRow(item);
+  }
+}
+
+QDataStream& operator<<(QDataStream& out,
+                        const RadioBrowserService::Stream& stream) {
+  out << stream.name_ << stream.url_;
+  return out;
+}
+
+QDataStream& operator>>(QDataStream& in, RadioBrowserService::Stream& stream) {
+  in >> stream.name_ >> stream.url_;
+  return in;
+}
+
+void RadioBrowserService::ReloadSettings() {
+  QSettings s;
+  s.beginGroup(kSettingsGroup);
+
+  main_server_url_ = s.value("server").toString();
+
+  if (root_ && root_->hasChildren()) {
+    root_->removeRows(0, root_->rowCount());
+    root_->setData(true, InternetModel::Role_CanLazyLoad);
+  }
+}
+
+bool RadioBrowserService::EnsureServerConfig() {
+  if (main_server_url_.isEmpty())
+    return false;
+  else
+    return true;
+}
+
+void RadioBrowserService::ShowConfig() {
+  app_->OpenSettingsDialogAtPage(SettingsDialog::Page_RadioBrowser);
+}
+
+void RadioBrowserService::Search(int search_id, const QString& query,
+                                 const int limit) {
+  QString determinedUrl =
+      RadioBrowserService::SearchUrl.arg(main_server_url_, query).arg(limit);
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting channels"));
+
+  NewClosure(reply, SIGNAL(finished()), this,
+             SLOT(SearchFinishedInternal(QNetworkReply*, int, int)), reply,
+             task_id, search_id);
```

Comment:
With qt5, you should be able to use a lambda in this kind of case. Take a look a the functor versions of the connect method: https://doc.qt.io/qt-5/qobject.html#connect-4

Reply:
I saw no connect() using the new syntax in the code, so i used the old one. Do you prefer me to convert them all and use lambdas where possible?

Reply:
In my option, it's cleaner, but it should achieve the same result. Dropbox has some examples:
https://github.com/clementine-player/Clementine/blob/master/src/internet/dropbox/dropboxservice.cpp#L179

Reply:
Done, switched to the new qt5 connect syntax and used lambda when possible.